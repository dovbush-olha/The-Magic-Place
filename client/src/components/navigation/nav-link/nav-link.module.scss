@import "../../../scss/settings";

.navLink {
  font-size: var(--14px);
  line-height: var(--18px);
  font-weight: font-weight(bold);
  color: color(text, beige);
  width: 120px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-bottom: 1px solid;
  border-image-source: linear-gradient(270deg, rgba(255 255 255 / 0%) 0%, #ffd480 46.87%, rgba(255 255 255 / 0%) 100%);
  border-image-slice: 1;
  position: relative;
  height: 40px;
  margin-bottom: 10px;

  &.active {
    &::before {
      position: absolute;
      content: "";
      inset: 7px;
      background-color: #b28f47;
      filter: blur(24px);
    }
  }
}
