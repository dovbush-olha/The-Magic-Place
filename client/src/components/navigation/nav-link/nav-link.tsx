import { NavLink as ReactRouterNavLink, NavLinkProps } from 'react-router-dom';

import styles from './nav-link.module.scss';

export function NavLink({ children, to, ...props }: NavLinkProps) {
  return (
    <ReactRouterNavLink
      className={styles.navLink}
      to={to}
      {...props}
    >
      {children}
    </ReactRouterNavLink>
  );
}
