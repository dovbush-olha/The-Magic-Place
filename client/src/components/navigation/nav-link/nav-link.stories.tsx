import { Meta, StoryObj } from '@storybook/react';

import { NavLink } from './nav-link';

const meta: Meta<typeof NavLink> = {
  title: 'Components/Navigation/NavLink',
  component: NavLink,
};

export default meta;

type Story = StoryObj<typeof NavLink>;

export const Default: Story = {
  args: {
    to: '/',
    children: 'Link',
  },
};
