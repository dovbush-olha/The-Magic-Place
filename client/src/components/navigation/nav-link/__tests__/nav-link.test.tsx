import { renderWithRouter } from '@/testing/test-utils.tsx';

import { NavLink } from '../nav-link.tsx';

describe('NavLink component', () => {
  it('renders correctly with children and to prop', () => {
    const { getByTestId } = renderWithRouter(<NavLink to="/home">Home</NavLink>);

    const navLink = getByTestId('nav-link');
    expect(navLink).toBeInTheDocument();
    expect(navLink).toHaveTextContent('Home');
  });

  it('passes the correct "to" prop', () => {
    const { getByTestId } = renderWithRouter(<NavLink to="/about">About</NavLink>);

    const navLink = getByTestId('nav-link');
    expect(navLink).toHaveAttribute('href', '/about');
  });

  it('does not render when children is missing', () => {
    const { container } = renderWithRouter(<NavLink to="/home" />);

    expect(container.firstChild).not.toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithRouter(<NavLink to="/active">Active Link</NavLink>);
    expect(asFragment()).toMatchSnapshot();
  });
});
