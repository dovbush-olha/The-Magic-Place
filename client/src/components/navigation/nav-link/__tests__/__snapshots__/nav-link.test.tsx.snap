// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavLink component > matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="_navLink_101612"
    data-testid="nav-link"
    href="/active"
  >
    Active Link
  </a>
</DocumentFragment>
`;
